docs/API.md

```diff
@@ -540,3 +549,59 @@ $ curl -G -d 'from=2015-03-10' -d 'to=2015-03-11' --url vmpooler.company.com/api
   ]
 }
 ```
+
+#### Changing configuration via API <a name="poolconfig"></a>
+
+##### POST /config/poolsize
```

Comment:
We could consider alternately using a single endpoint. I started with poolsize and then added pooltemplate. Part of my goal here has been to minimize complexity so I thought having two separate paths under /config seemed reasonable for that. With that said, I'm open to change it if it doesn't make sense.

As of right now I hadn't made a plan to add pools that aren't configured. If it's something we want to support we could add it. My thought is that it allows the configuration file to serve as a source of truth about which pools are configured, even though size and template configurations may change.

I think it would be possible to add pools live, but haven't determined exactly what additional work would be required to do it. In the cases setup now there's already a check_pool thread running for each pool that exists to detect and deploy changes. Something would have to spawn a new check_pool thread for the new pool when deploying it, which may add some complication, but is likely possible.

Reply:
Yeah that makes sense, a single endpoint would be nice to have in the future (what's here makes sense and is great too). Do you want me to file a VMPOOLER ticket to consider the single endpoint?

Making the config. file the source of truth definitely makes sense, but from an automation standpoint it would be great if at least a simple iteration of something like adding pools live would go out during the insurgency and only be available on vmpooler-dev (one idea to do this is to introduce a new config. entry 'turn on experimental features: true' that the production instances would not have, but the dev pooler would). What I'm thinking is then the vmpooler-dev. part would consist of (in CI code)

```
(1) Make an API request to add the new pool. You'd know the template and pool size beforehand.
(2) Poll the ready VMs endpoint until size VMs have been moved to Ready. This can fail after 15-20 minutes, since 15-20 minutes is about the time vmpooler stops checking for Ready VMs.
(3) Check-out a ready VM and run some simple smoke tests on it (e.g. ensure you can connect to it as a user, for a start)
(4) Destroy the ready VM via. an API request
(5) Verify that the ready VM has been destroyed
(6) Delete the newly created pool from the config., since it is no longer needed.
(7) Make a PR to puppetlabs-modules adding the new pool to the production instances. Deploying to production can be manual after this part, until we've improved the automation here.
```

I'm thinking the dev pooler wouldn't have any pools, only creating pools when we're about to test an image (and the size for that pool would always be 1).

A single config. endpoint might make this easier (b/c then you can introduce the new pool as part of the config.).

Reply:
It's perhaps a little strange, but couldn't you accomplish this workflow by simply treating a pool title as arbitrary and swapping out its template path? I acknowledge this may seem a little odd, but from vmpooler's perspective I don't think it knows any difference, so in that sense you could have something generic and swap out its template as needed.

Please do create a ticket in the VMPOOLER project for the single endpoint. I'd like to get some input on it and it will help to track the request.

Reply:
Ah yes, we could. Thanks for the suggestion! I think [cricket.pp](https://github.com/puppetlabs/puppetlabs-modules/blob/production/site/profile/manifests/vmpooler/pools/cricket.pp) might need to be updated. I'm thinking of removing all of those pools and just having a single `dev-pool` entry defaulting to a template we know exists w/ size 1. Then we only need to update the template. Would that work/be fine?

Sure, will go ahead and do that.

Reply:
Filed https://tickets.puppetlabs.com/browse/POOLER-110